src/pybind/mgr/rgw/module.py

```diff
@@ -349,3 +349,54 @@ def shutdown(self) -> None:
         self.log.info('Stopping')
         self.run = False
         self.event.set()
+
+    def list_all_realm_tokens(self) -> List[Dict]:
+        realms_info = []
+        for realm_info in RGWAM(self.env).get_realms_info():
+            if not realm_info['master_zone_id']:
+                realms_info.append({'realm': realm_info['realm_name'], 'token': 'realm has no master zone'})
+            elif not realm_info['endpoint']:
+                realms_info.append({'realm': realm_info['realm_name'], 'token': 'master zone has no endpoint'})
+            elif not (realm_info['access_key'] and realm_info['secret']):
+                realms_info.append({'realm': realm_info['realm_name'], 'token': 'master zone has no access/secret keys'})
+            else:
+                keys = ['realm_name', 'realm_id', 'endpoint', 'access_key', 'secret']
+                realm_token = RealmToken(**{k: realm_info[k] for k in keys})
+                realm_token_b = realm_token.to_json().encode('utf-8')
+                realm_token_s = base64.b64encode(realm_token_b).decode('utf-8')
+                realms_info.append({'realm': realm_info['realm_name'], 'token': realm_token_s})
```

Comment:
I like the idea of having this as  a function but please, can you update the code on the method `def list_realm_tokens(self) -> HandleCommandResult:` to use the new function you have created instead of duplicating the code?

Reply:
Thanks @rkachach , I have made the required changes